src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -281,10 +281,20 @@ export class IndexPatternsService {
     options: GetFieldsOptions,
     fieldAttrs: FieldAttrs = {}
   ) => {
-    const scriptedFields = Object.values(fields).filter((field) => field.scripted);
+    const fieldsAsArr = Object.values(fields);
+    const scriptedFields = fieldsAsArr.filter((field) => field.scripted);
     try {
+      let updatedFieldList: FieldSpec[];
       const newFields = (await this.getFieldsForWildcard(options)) as FieldSpec[];
-      return this.fieldArrayToMap([...newFields, ...scriptedFields], fieldAttrs);
+      /*
+       * Only update field list if field caps finds fields. To support beats creating index pattern and dashboard before docs
+       */
+      if (newFields.length) {
```

Comment:
I think the error that happens in `this.getFieldsForWildcard` when there are no matching indices will happen before this code has a chance to run.  Perhaps we can move this into the catch statement where it checks for `IndexPatternMissingIndices`.